backend/dal/Services/Admin/Concrete/ProjectService.cs

```diff
@@ -178,19 +178,22 @@ public void Add(IEnumerable<Project> projects)
             projects.ThrowIfNull(nameof(projects));
             this.User.ThrowIfNotAuthorized(Permissions.SystemAdmin, Permissions.AgencyAdmin);
 
-            projects.ForEach((project) =>
+            if (projects.Any())
             {
-                if (project == null) return;
-
-                if (String.IsNullOrWhiteSpace(project.ProjectNumber))
+                projects.ForEach((project) =>
                 {
-                    project.ProjectNumber = $"TEMP-{DateTime.UtcNow.Ticks:00000}";
-                }
+                    if (project == null) return;
 
-                this.Context.Projects.Add(project);
-            });
+                    if (String.IsNullOrWhiteSpace(project.ProjectNumber))
+                    {
+                        project.ProjectNumber = $"TEMP-{DateTime.UtcNow.Ticks:00000}";
```

Comment:
I know this isn't new but out of curiosity, is it pretty common to use `DateTime` to generate unique identifiers?

Reply:
I haven't really seen this before but that is just me. I've seen random GUIDS which is kinda similar.

Reply:
This isn't a perfect solution, as it is theoretically possible for more than one project to receive the same `tick`.  Although it's unlikely, especially in relation to our architecture and use-case.   Having an incrementing number is generally friendlier than a GUID.  It also sorts far better.